base/dict.jl

```diff
@@ -189,6 +189,16 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict, [element_type=Pair{K,V}])
```

Comment:
Are you suggesting to write `similar(d::Dict, [element_type=eltype(d)])`? One is going to ask why the `eltype(d)` needs to be given if `d` is passed already. On the other hand, also `similar(d::Dict, [element_type=eltype(e::Dict)])` is not correct, because the method is valid for a superclass of dictionaries.